test/handler/pool_test.exs

```diff
@@ -283,18 +275,33 @@ defmodule Handler.PoolTest do
       {root, composed} = setup_composed_pools()
 
       opts = [max_ms: 200, max_heap_bytes: 2 * 1024, task_name: task_name]
-      {:ok, _ref} = Pool.async(composed, fn -> :timer.sleep(60_000) end, opts)
+      {:ok, ref} = Pool.async(composed, fn -> :timer.sleep(60_000) end, opts)
+      :sys.get_state(composed)
 
       {:ok, 1} = Pool.kill(composed, task_name)
+      assert {:error, exception} = Pool.await(ref)
+
+      assert exception == %Handler.ProcessExit{
+               reason: :user_killed,
+               message: "User killed the process"
+             }
+
```

Comment:
We did similar changes to this test(it was very flawed before) but one thing I don't see on this PR is an assertion that the task did in fact die. One of the issues that was happening when using this was the because Task.async had it's exited trapped, it would never really die, but we would return to the client the proper ProcessExit message.

Can you add refute Process.alive?(pid) here?